tests/frontier/scenarios/common.py

```diff
@@ -0,0 +1,302 @@
+    CALL_VALUE = 6
+    CALL_DATALOAD_0 = 7
+    CALL_DATASIZE = 8
+    GASPRICE = 9
+    BLOCKHASH_0 = 10
+    COINBASE = 11
+    TIMESTAMP = 12
+    NUMBER = 13
+    GASLIMIT = 14
+
+
+@dataclass
+class ProgramResult:
+    """
+    Describe expected result of a program.
+
+    Attributes:
+        result (int | ScenarioExpectOpcode): The result of the program
+        from_fork (Fork): The result is only valid from this fork (default: Frontier)
+        static_support (bool): Can be verified in static context (default: True)
+
+    """
+
+    result: int | ScenarioExpectOpcode
+
+    """The result is only valid from this fork"""
+    from_fork: Fork = Frontier
+    static_support: bool = True
+
+
+@dataclass
+class ScenarioEnvironment:
+    """
+    Scenario evm environment
+    Each scenario must define an environment on which program is executed
+    This is so post state verification could check results of evm opcodes.
+    """
+
+    code_address: Address  # Op.ADDRESS, address scope for program
+    code_caller: Address  # Op.CALLER, caller of the program
+    selfbalance: int  # Op.SELFBALANCE, balance of the environment of the program
+    ext_balance: int  # Op.BALANCE(external) of fixed address deployed in state
+    call_value: int  # Op.CALLVALUE of call that is done to the program
+    call_dataload_0: int  # Op.CALLDATALOAD(0) expected result
+    call_datasize: int  # Op.CALLDATASIZE expected result
+    has_static: bool = False  # Weather scenario execution context is static
+
+
+@dataclass
+class ExecutionEnvironment:
+    """Scenario execution environment which is determined by test."""
+
+    fork: Fork
+    gasprice: int
+    origin: Address
+    coinbase: Address
+    timestamp: int
+    number: int
+    gaslimit: int
+
+
+@dataclass
+class ScenarioGeneratorInput:
+    """
+    Parameters for the scenario generator function.
+
+    Attributes:
+        fork (Fork): Fork for which we ask to generate scenarios
+        pre (Alloc): Access to the state to be able to deploy contracts into pre
+        operation (Bytecode): Evm bytecode program that will be tested
+        external_address (Address): Static external address for ext opcodes
+
+    """
+
+    fork: Fork
+    pre: Alloc
+    operation_code: Bytecode
+    external_address: Address
+    external_balance: int
+
+
+@dataclass
+class Scenario:
+    """
+    Describe test scenario that will be run in test for each program.
+
+    Attributes:
+        name (str): Scenario name for the test vector
+        code (Address): Address that is an entry point for scenario code
+        env (ScenarioEnvironment): Evm values for ScenarioExpectAddress map
+        reverting (bool): If scenario reverts program execution, making result 0 (default: False)
+
+    """
+
+    name: str
+    code: Address
+    env: ScenarioEnvironment
+    halts: bool = False
+
+
+def translate_result(
+    res: ProgramResult, env: ScenarioEnvironment, exec_env: ExecutionEnvironment
+) -> int:
```

Comment:
```suggestion
) -> int | Address:
```
So there won't be any need to transform the addresses to `int`.